### PR TITLE
fixed macOS build to use OpenJPEG and to test all combinations of PNG, JPEG options

### LINKED ISSUE #221: MacOS workflow uses invalid option: ENABLE_OpenJPEG is not valid cmake option
```
CMake Warning:
  Manually-specified variables were not used by the project:

    ENABLE_OpenJPEG
```

https://github.com/NOAA-EMC/NCEPLIBS-g2c/blob/c0e92eea86541ae54ccb5137e3bc75a627eee11b/.github/workflows/macOS.yml#L51

https://github.com/NOAA-EMC/NCEPLIBS-g2c/blob/c0e92eea86541ae54ccb5137e3bc75a627eee11b/.github/workflows/linux.yml#L52

It should be `USE_OpenJPEG`

### PR DESCRIPTION
**Pull request by edwardhartnett**
Fixes #221

### CODE REVIEW

File: .github/workflows/macOS.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
-name: macOS Build
-on: [push, pull_request]
+name: macOS-build
+on: [pull_request push]
```
Comment (DusanJovic-NOAA) [inline .github/workflows/macOS.yml]: Missing comma. Why is the order changed?

File: .github/workflows/macOS.yml [hunk 2]
```diff
@@ -8,6 +8,35 @@ jobs:
       CC: gcc-11
       CXX: g++-11
```
Comment (DusanJovic-NOAA) [inline .github/workflows/macOS.yml]: No need to set CXX variable. On MacOS I suggest we test this library using native compiler (clang).

File: .github/workflows/macOS.yml [hunk 3]
```diff
@@ -48,10 +77,12 @@ jobs:
...
-        cmake -DJasper_ROOT=~/Jasper -DENABLE_DOCS=ON -DENABLE_OpenJPEG=ON ..
+        cmake ${{ matrix.config.options }} ..
         make -j2
 
     - name: test
       run: |
         cd $GITHUB_WORKSPACE/g2c/build
         ctest  --output-on-failure
+
+
```
Comment (DusanJovic-NOAA) [inline .github/workflows/macOS.yml]: Remove trailing empty lines.

File: .github/workflows/macOS.yml (hunk 1 above)
Comment (edwardhartnett) [reply .github/workflows/macOS.yml]: Experimentation. I will change it back, though it seems to work fine. ;-)

File: .github/workflows/macOS.yml (hunk 3 above)
Comment (edwardhartnett) [reply .github/workflows/macOS.yml]: OK

File: .github/workflows/macOS.yml (hunk 2 above)
Comment (edwardhartnett) [reply .github/workflows/macOS.yml]: OK

**Review by DusanJovic-NOAA — APPROVED** [review]

**Review by BrianCurtis-NOAA — APPROVED** [review]

**Pull request merged**